cmd/gapit/memory.go

```diff
@@ -142,81 +149,81 @@ func (verb *memoryVerb) Run(ctx context.Context, flags flag.FlagSet) error {
 					info.XOffset, info.YOffset)
-				fmt.Printf("\t\tBlock Extent: (%v, %v)\n",
+				fmt.Fprintf(w, "\t\tBlock Extent: (%v, %v)\n",
 					info.Width, info.Height)
-				fmt.Printf("\t\tMip Level:    %v\n", info.MipLevel)
-				fmt.Printf("\t\tArray Layer:  %v\n", info.ArrayLayer)
-				if asp := stringifyAspects(info.Aspects); len(asp) > 0 {
-					fmt.Printf("\t\tAspects:      %v\n", asp)
-				}
+				fmt.Fprintf(w, "\t\tMip Level:    %v\n", info.MipLevel)
+				fmt.Fprintf(w, "\t\tArray Layer:  %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tAspects:      %v\n", aspectList(info.Aspects))
 			case *api.MemoryBinding_SparseImageMetadata:
 				info := val.SparseImageMetadata
-				fmt.Printf("\t\tArray Layer:     %v\n", info.ArrayLayer)
-				fmt.Printf("\t\tMip Tail Offset: %v\n", info.Offset)
+				fmt.Fprintf(w, "\t\tArray Layer:     %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tMip Tail Offset: %v\n", info.Offset)
 			case *api.MemoryBinding_SparseImageMipTail:
 				info := val.SparseImageMipTail
-				fmt.Printf("\t\tArray Layer:     %v\n", info.ArrayLayer)
-				fmt.Printf("\t\tMip Tail Offset: %v\n", info.Offset)
-				if asp := stringifyAspects(info.Aspects); len(asp) > 0 {
-					fmt.Printf("\t\tAspects:      %v\n", asp)
-				}
+				fmt.Fprintf(w, "\t\tArray Layer:     %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tMip Tail Offset: %v\n", info.Offset)
+				fmt.Fprintf(w, "\t\tAspects:         %v\n", aspectList(info.Aspects))
 			case *api.MemoryBinding_SparseOpaqueImageBlock:
-				fmt.Printf("\t\tImage Memory Offset: %v\n",
+				fmt.Fprintf(w, "\t\tImage Memory Offset: %v\n",
 					val.SparseOpaqueImageBlock.Offset)
 			case *api.MemoryBinding_SparseBufferBlock:
-				fmt.Printf("\t\tBuffer Memory Offset: %v\n",
+				fmt.Fprintf(w, "\t\tBuffer Memory Offset: %v\n",
 					val.SparseBufferBlock.Offset)
 			}
 		}
 
 		aliases := bindings.computeAliasing()
 		if len(aliases) == 0 {
-			fmt.Println("\tNo aliased regions")
+			fmt.Fprintln(w, "\tNo aliased regions")
 		} else {
-			fmt.Printf("\t%v aliased regions:\n", len(aliases))
+			fmt.Fprintf(w, "\t%v aliased regions:\n", len(aliases))
 			for i, a := range aliases {
-				fmt.Printf("\t%v:\n", i)
-				fmt.Println("\t\tOffset: ", a.offset)
-				fmt.Println("\t\tSize:   ", a.size)
-				fmt.Println("\t\tShared by:")
+				fmt.Fprintf(w, "\t%v:\n", i)
+				fmt.Fprintln(w, "\t\tOffset: ", a.offset)
+				fmt.Fprintln(w, "\t\tSize:   ", a.size)
+				fmt.Fprintln(w, "\t\tShared by:")
 				for _, s := range a.sharers {
-					fmt.Printf("\t\t\t%v\n", s)
+					fmt.Fprintf(w, "\t\t\t%v\n", s)
 				}
 			}
 		}
 	}
+	w.Flush()
 	return nil
 }
 
+type aspectList []api.AspectType
+
+func (a aspectList) Format(f fmt.State, c rune) {
+	if len(a) == 0 {
+		return
+	}
+	strs := make([]string, len(a))
+	for i, asp := range a {
+		var typ string
+		switch asp {
+		case api.AspectType_COLOR:
+			typ = "Color"
+		case api.AspectType_DEPTH:
+			typ = "Depth"
+		case api.AspectType_STENCIL:
+			typ = "Stencil"
```

Comment:
Please add an unknown case that includes the integer value

Reply:
`api.AspectType` is an enum, so I'm not sure how to include the integer value there.  I can add an unknown type to the enum though.

Reply:
Something like:

```go
func (a AspectType) Format(f fmt.State, c rune) {
	switch a {
	case api.AspectType_COLOR:
		fmt.Fprint(f, "Color")
	case api.AspectType_DEPTH:
		fmt.Fprint(f, "Depth")
	case api.AspectType_STENCIL:
		fmt.Fprint(f, "Stencil")
	default:
		fmt.Fprintf(f, "Unknown AspectType %d", (int)a)
	}
}
```

Of course protobuf already produces a `String()` method for you, but I'm guessing it's not what you want?

Reply:
Oh, I didn't realize protobuf enums could take values that weren't specified in the proto description, but I suppose it makes sense it would for version compatibility.